Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -2483,7 +2483,8 @@ bool vtkMRMLSliceLogic::IsEventInsideVolume(bool background, double worldPos[3])
   volumeNode->GetImageData()->GetExtent(volumeExtent);
   for (int i = 0; i < 3; i++)
     {
-    if (ijkPos[i]<volumeExtent[i * 2] || ijkPos[i]>volumeExtent[i * 2 + 1])
+    double epsilon = 1e-6;
```

Comment:
In VTK, the voxel coordinate refers to the center of the voxel and so the image bounds go beyond the position of the first and last voxels by half voxel. Therefore, instead of an "epsilon", we should actually use half voxel (absolute value of 0.5, since we are already in voxel coordinates). Please also fix the missing spaces around the `<` and `>` operators.

Reply:
Thanks for the review. I have updated the code with the half voxel shift and included a code comment with your description so it is clear in the future what the 0.5 usage means. I have tested with the problematic scene and things are still working. I have also fixed the missing spaces around the operators.